### DISCUSSION TITLE
Random Values from WeibullAFT?

### DISCUSSION THREAD

**Discussion by edwardreed81** [CamDavidsonPilon/lifelines#1218] [Q&A]
Is it possible to draw random values from the distributions fitted by `WeibullAFTFitter`?

Something along these lines:
```python
from lifelines import WeibullAFTFitter
from lifelines.datasets import load_rossi

rossi = load_rossi()

aft = WeibullAFTFitter()
aft.fit(rossi, duration_col='week', event_col='arrest')
aft.rvs(covariates='prio', values=[0, 3, 6], size=100)
```
to get an array of shape `(3, 100)` of random values pulled from the distributions defined by `values`.

I can do this discretely using `predict_survival_function` and linear interpolation on the ECDF, but that doesn't allow values higher than `timeline.max()`, which seems like something possible using a parametric model.

Is there a way to do this currently? If not, is it possible?

**Comment by edwardreed81** [comment]
After more reading and experimentation, I've figured out how to draw samples.

My simplistic example:
```python
from scipy.stats import weibull_min
B0 = aft.params_['lambda_', 'Intercept']
B1 = aft.params_['lambda_', 'prio']

value = 0

rho = np.exp(aft.params_['rho_', 'Intercept'])

lam = np.exp(B0 + B1*value)

w = weibull_min.rvs(c=rho, scale=lam, size=1000)
```
Obviously this only accounts for a single covariate and is not general at all.

Might be nice if this functionality was built in?

**Comment by CamDavidsonPilon** [comment]
Hi @edwardreed81,
It's not possible atm, but I think it's a nice idea, though! Can I ask what you are interested in doing with the samples? 

`aft.rvs(covariates='prio', values=[0, 3, 6], size=100)`

So ^ will vary prio between 0, 3 and 6? What values do you set the other covariates too (or is like `plot_partial_effects_on_outcome` and we set the other covariates to their median/mode value)?

**Reply by edwardreed81** [reply]
I'm using the samples for doing a lot of predictions and caching a lot of samples and pulling from the cache is far faster than generating them on the fly.

And yes, I figure it could work just like `plot_partial_effects_on_outcome`. Or it could require a value for each covariate, doesn't really matter.
